softgroup/basis/common-router/common-router-impl/src/main/java/com/softgroup/common/router/impl/router/factory/RootFactoryImpl.java

```diff
@@ -0,0 +1,15 @@
+package com.softgroup.common.router.impl.router.factory;
+
+import com.softgroup.common.router.api.CommonRouterHandler;
+import com.softgroup.common.router.api.factory.RootFactory;
+import com.softgroup.common.router.api.factory.RouterFactory;
+import org.springframework.stereotype.Component;
+
+/**
+ * Created by nikmlk on 07.03.17.
+ */
+
+@Component
+public class RootFactoryImpl<T extends CommonRouterHandler> extends RouterFactory<T> implements RootFactory<T> {
```

Comment:
@melnyknikolay Why you use generic parameter T ? I think this class should use defined type CommonRouterHandler.